coqui-tts-gui.py

```diff
@@ -0,0 +1,202 @@
+            return
+    generatebutton.config(state="disabled")
+    exportbutton.config(state="disabled")
+    model_path = None
+    config_path = None
+    vocoder_path = None
+    vocoder_config_path = None
+    path = Path(__file__).parent / "TTS/.models.json"
+    manager = ModelManager(path)
+    model_name = 'tts_models/' + ttsmodelbox.get()
+    print(f'model_name is {model_name}')
+    # for dev
+    model_path, config_path, model_item = manager.download_model(model_name)
+    # for master
+    #model_path, config_path = manager.download_model(model_name)
+    vocoder_name = 'vocoder_models/' + vocodermodelbox.get()
+    print(f'vocoder_name is {vocoder_name}')
+    # for dev
+    vocoder_path, vocoder_config_path, model_item = manager.download_model(vocoder_name)
+    # for master
+    #vocoder_path, vocoder_config_path = manager.download_model(vocoder_name)
+    synthesizer = Synthesizer(model_path, config_path, vocoder_path, vocoder_config_path, cudacheckbutton.instate(['selected']))
+    wav = synthesizer.tts(inputbox.get("1.0","end-1c"))
+    synthesizer.save_wav(wav, str(f.name))
+    generatebutton.config(state="enabled")
+    exportbutton.config(state="enabled")
+    print("All done!")
+
+def opentext():
+    file = filedialog.askopenfile(mode='rt', defaultextension=".txt", filetypes = [("Text files", ".txt")])
+    if file is None: # asksaveasfile return `None` if dialog closed with "cancel".
+        return
+    contents = file.read()
+    inputbox.insert('1.0',contents)
+    inputbox.edit_modified(False)
+    file.close()
+
+def checkopentext():
+    if inputbox.edit_modified() == True:
+        response=messagebox.askyesnocancel(message = "You have unsaved changes. Do you want to save?")
+        if response == True:
+            savetextandopen()
+        elif response == False:
+            opentext()
+        else:
+            return
+    else:
+        opentext()
+    
+# Creating tkinter window
+window = tk.Tk()
+window.geometry('')
+window.title("Coqui TTS GUI")
+window.resizable(False, False)
+# Label 
+ttk.Label(window, text = "Enter text here",  
+        font = ("Tahoma", 10)).grid(column = 0, columnspan = 3,  
+        row = 12, padx = 10, pady = 12)
+
+# Text
+inputbox = scrolledtext.ScrolledText(window, height = 15, width = 70, undo = True)
+inputbox.grid(column = 0, columnspan = 3, 
+        row = 13, padx = 10, pady = 12)
+
+# Label 
+ttk.Label(window, text = "Select the tts_model",  
+        font = ("Tahoma", 10)).grid(column = 0,  
+        row = 14, padx = 10, pady = 12) 
+  
+n = tk.StringVar() 
+ttsmodelbox = ttk.Combobox(window, width = 32, textvariable = n, state = "readonly") 
+# Adding combobox drop down list 
+ttsmodelbox['values'] = ('en/ljspeech/glow-tts',  
+                          'en/ljspeech/tacotron2-DCA', 
+                          'en/ljspeech/speedy-speech-wn', 
+                          'es/mai/tacotron2-DDC', 
+                          'fr/mai/tacotron2-DDC') 
```

Comment:
read the models from ```.models.json``` instead of hard code.

Reply:
How can I? https://stackoverflow.com/questions/59760156/how-to-append-values-from-json-file-to-tkinter-combobox-in-python